webapp-django/crashstats/crashstats/views.py

```diff
@@ -2063,35 +2061,34 @@ def report_list(request, partial=None, default_context=None):
                     '%Y-%m-%d %H:%M:%S'
                 )
 
-        if os_count:
-            correlation_os = max(os_count.iterkeys(),
-                                 key=lambda k: os_count[k])
-        else:
-            correlation_os = None
-        context['correlation_os'] = correlation_os
-
-        if version_count:
-            correlation_version = max(version_count.iterkeys(),
-                                      key=lambda k: version_count[k])
-        else:
-            correlation_version = None
-        if correlation_version is None:
-            correlation_version = ''
-        context['correlation_version'] = correlation_version
+        # arbitrarily, pick the X most popular OS and version combinations
+        context['correlation_combos'] = [
+            {
+                'os': k[0],
+                'version': k[1],
+            }
+            for k, v in sorted(
+                counts.items(),
+                key=lambda x: x[1],
+                reverse=True
+            )
+        ][:settings.MAX_CORRELATION_COMBOS_PER_SIGNATURE]
```

Comment:
Sorry for the messy list comprehension. It might make @lonnen throw up in his mouth a little. 

The more important point here is the crazy fact that we're picking an arbitrary number. Why 3? Why not 2 or 7 or 34? I don't know. I just didn't want to cause too much AJAX queries if there's a signature that has many different kinds of versions and OSes. 

Having said that, this whole logic is based on the **first page** of the report list. (You made this point yourself Adrian). So if a signature has 300 44.0a2 on the first page and 43.0b1 only on the second page, you might miss out on valuable correlations.

What to do? Add a new middleware endpoint thing where you can get aggregates of crashes for that signature and pick the 3 most "crashiest". 
Or, do that with a SuperSearch query? Or, just accept this as much-better-than-it-was-and-it-ll-soon-be-replaced-by-signature-report-anyway.